kale/pipeline/mpca_trainer.py

```diff
@@ -0,0 +1,159 @@
+# =============================================================================
+# Author: Shuo Zhou, szhou20@sheffield.ac.uk
+#         Haiping Lu, h.lu@sheffield.ac.uk or hplu@ieee.org
+# =============================================================================
+
+"""Implementation of MPCA->Feature Selection->Linear SVM/LogisticRegression Pipeline
+"""
+
+import logging
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.feature_selection import f_classif
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+from sklearn.utils.validation import check_is_fitted
+
+from ..embed.mpca import MPCA
+
+classifiers = {"svc": [SVC, {"kernel": ["linear"], "C": np.logspace(-3, 2, 6)}],
```

Comment:
Have you checked the Matlab code options? 
Does this cover the options used in our CMR Matlab code? It seems that we used the default there, which seems to be 1/n (an adaptive value) according to https://uk.mathworks.com/help/stats/fitclinear.html#d123e312449
At least this value of `C` should be covered.

Reply:
There is a close value in `np.logspace(-3, 2, 6)`, which is a list `[0.001, 0.01, 0.1, 0, 1, 10]`. The optimal value of `C` will be determined by grid search if `classifier_params="auto"`. I will add 1/n to this list if necessary.

Reply:
@sz144 1/n is an interesting and smart choice because when n --> infinity, 1/n --> 0 and no regularisation is needed.